frontend/app/src/components/account-management/PremiumCredentials.vue

```diff
@@ -32,87 +32,68 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent, ref, toRefs, watch } from 'vue';
-import { IVueI18n } from 'vue-i18n';
+<script setup lang="ts">
+import { toRefs, watch } from 'vue';
+import { useI18n } from 'vue-i18n-composable';
 import RevealableInput from '@/components/inputs/RevealableInput.vue';
-import i18n from '@/i18n';
 import { trimOnPaste } from '@/utils/event';
 
-const setupValidationRules = (i18n: IVueI18n) => {
-  const apiKeyRules = [
-    (v: string) =>
-      !!v || i18n.t('premium_credentials.validation.non_empty_key').toString()
-  ];
-  const apiSecretRules = [
-    (v: string) =>
-      !!v ||
-      i18n.t('premium_credentials.validation.non_empty_secret').toString()
-  ];
-  return { apiKeyRules, apiSecretRules };
-};
+const { t, tc } = useI18n();
+const apiKeyRules = [
+  (v: string) =>
+    !!v || t('premium_credentials.validation.non_empty_key').toString()
+];
+const apiSecretRules = [
+  (v: string) =>
+    !!v || t('premium_credentials.validation.non_empty_secret').toString()
+];
 
-const PremiumCredentials = defineComponent({
-  name: 'PremiumCredentials',
-  components: { RevealableInput },
-  props: {
-    loading: { required: true, type: Boolean },
-    enabled: { required: true, type: Boolean },
-    apiSecret: { required: true, type: String },
-    apiKey: { required: true, type: String },
-    syncDatabase: { required: true, type: Boolean }
-  },
-  emits: ['update:api-key', 'update:api-secret', 'update:sync-database'],
-  setup(props, { emit }) {
-    const { enabled } = toRefs(props);
-    const showKey = ref(false);
-    const showSecret = ref(false);
+const props = defineProps({
+  loading: { required: true, type: Boolean },
+  enabled: { required: true, type: Boolean },
+  apiSecret: { required: true, type: String },
+  apiKey: { required: true, type: String },
+  syncDatabase: { required: true, type: Boolean }
+});
 
-    const updateApiKey = (apiKey: string) => {
-      emit('update:api-key', apiKey?.trim() ?? '');
-    };
+const emit = defineEmits([
+  'update:api-key',
```

Comment:
no need to do in this PR but better prefer the other annotation
`defineEmits<{(e: 'update:api-key', value: string):void}>()`

as it provides better type-checking in the emitters